### PR TITLE
[shared_preferences] fix cast error and mutable list error with `getStringList`

### PR DESCRIPTION
**Pull request by tarrinneal**
fixes https://github.com/flutter/flutter/issues/153106
fixes https://github.com/flutter/flutter/issues/153112

Will add a follow up pr adding a test to the top level package after this pr lands.

### CODE REVIEW

File: packages/shared_preferences/shared_preferences_windows/example/integration_test/shared_preferences_test.dart [hunk 1]
```diff
@@ -397,6 +400,24 @@ void main() {
       await preferences.setStringList(listKey, testList, emptyOptions);
       expect(await preferences.getStringList(listKey, emptyOptions), testList);
     });
+    testWidgets('getStringList does not throw cast error',
```
Comment (stuartmorgan) [inline packages/shared_preferences/shared_preferences_windows/example/integration_test/shared_preferences_test.dart:403]: Nit: blank line above this.

**Review by stuartmorgan — APPROVED** [review]
LGTM for the version that didn't add the deliberate failure.

**Comment by paulober** [comment]
Thanks. When can I expect the fix to release?

**Comment by stuartmorgan** [comment]
> Thanks. When can I expect the fix to release?

Three days ago; we use a continuous release model for the packages in this repository. You just need to `flutter pub upgrade` to update your transitive dependencies.

**Pull request merged**